core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\Voter\DataContainer;
+
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\CoreBundle\Security\DataContainer\ReadAction;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\CacheableVoterInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+
+/**
+ * @internal
+ */
+class ArticleContentVoter implements VoterInterface, CacheableVoterInterface
+{
+    public function __construct(private readonly AccessDecisionManagerInterface $accessDecisionManager)
+    {
+    }
+
+    public function supportsAttribute(string $attribute): bool
+    {
+        return ContaoCorePermissions::DC_PREFIX.'tl_content' === $attribute;
+    }
+
+    public function supportsType(string $subjectType): bool
+    {
+        return ReadAction::class === $subjectType;
+    }
+
+    /**
+     * This only implements read access permission on tl_content
+     * to disable the "children" operation on tl_article. It should be extended
+     * to also check all other permissions on tl_content.
+     */
+    public function vote(TokenInterface $token, $subject, array $attributes): int
+    {
+        foreach ($attributes as $attribute) {
+            if (
+                !$subject instanceof ReadAction
+                || 'tl_article' !== ($subject->getCurrent()['ptable'] ?? null)
```

Comment:
This should be checked outside of the `foreach` loop, shouldn‘t it?

Reply:
technically yes, but all voter handle each attribute separately. There is always only one attribute, it's just an array for legacy reasons (also see https://github.com/symfony/symfony/blob/7.1/src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php#L54-L57).

Reply:
There is no way a static analyzer could ever understand this special case, so either we move the check outside the `foreach` loop or we remove the `foreach` loop entirely and work with `$attributes[0]` instead.

Reply:
the static analyzer does not seem to care about this, all build checks are green?

Reply:
If you increase the PHPStan level, this will be an error. So please change it as outlined above.